File: docs/code-examples/all/manual_indicator.cpp
// Shows how to manually associate one or more indicator components with arbitrary data.

#include <rerun.hpp>

#include <vector>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_manual_indicator");
    rec.spawn().exit_on_failure();

    std::vector<rerun::Position3D> positions = {
        {0.0, 0.0, 0.0},
        {10.0, 0.0, 0.0},
        {0.0, 10.0, 0.0},
    };
    std::vector<rerun::Color> colors = {
        {255, 0, 0},
        {0, 255, 0},
        {0, 0, 255},
    };
    std::vector<rerun::Radius> radii = {1.0};

    // Specify both a Mesh3D and a Points3D indicator component so that the data is shown as both a
    // 3D mesh _and_ a point cloud by default.
    rec.log(
        "points_and_mesh",
        rerun::Points3D::IndicatorComponent(),
        rerun::Mesh3D::IndicatorComponent(),
        positions,
        colors,
        radii
    );
}

File: docs/snippets/all/manual_indicator.py
"""Shows how to manually associate one or more indicator components with arbitrary data."""
import rerun as rr

rr.init("rerun_example_manual_indicator", spawn=True)

# Specify both a Mesh3D and a Points3D indicator component so that the data is shown as both a
# 3D mesh _and_ a point cloud by default.
rr.log(
    "points_and_mesh",
    [
        rr.Points3D.indicator(),
        rr.Mesh3D.indicator(),
        rr.components.Position3DBatch([[0.0, 0.0, 0.0], [10.0, 0.0, 0.0], [0.0, 10.0, 0.0]]),
        rr.components.ColorBatch([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]),
        rr.components.RadiusBatch([1.0]),
    ],
)
